.cbmc-batch/jobs/aws_cryptosdk_enc_ctx_size/aws_cryptosdk_enc_ctx_size_harness.c

```diff
@@ -31,22 +31,29 @@ void hash_iterator_generator(struct aws_hash_iter *new_iter, const struct aws_ha
 void hash_iterator_generator2(struct aws_hash_iter *new_iter, const struct aws_hash_iter *old_iter) {
     (void)old_iter;
     if (new_iter->status == AWS_HASH_ITER_STATUS_READY_FOR_USE) {
-        new_iter->element.key   = malloc(sizeof(struct aws_string));
+        new_iter->element.key = malloc(sizeof(struct aws_string));
+        __CPROVER_assume(new_iter->element.key != NULL);
         new_iter->element.value = malloc(sizeof(struct aws_string));
+        __CPROVER_assume(new_iter->element.value != NULL);
     }
 }
 
 void aws_cryptosdk_enc_ctx_size_harness() {
-    struct aws_hash_table map;
-    ensure_allocated_hash_table(&map, MAX_TABLE_SIZE);
-    __CPROVER_assume(aws_hash_table_is_valid(&map));
-    ensure_hash_table_has_valid_destroy_functions(&map);
+    /* Nondet Input */
+    struct aws_hash_table *map = can_fail_malloc(sizeof(*map));
+    size_t *size = can_fail_malloc(sizeof(size));
 
+    /* Assumptions */
+    __CPROVER_assume(map != NULL);
+    ensure_allocated_hash_table(map, MAX_TABLE_SIZE);
+    __CPROVER_assume(aws_hash_table_is_valid(map));
+    ensure_hash_table_has_valid_destroy_functions(map);
     size_t empty_slot_idx;
-    __CPROVER_assume(aws_hash_table_has_an_empty_slot(&map, &empty_slot_idx));
+    __CPROVER_assume(aws_hash_table_has_an_empty_slot(map, &empty_slot_idx));
+    __CPROVER_assume(size != NULL);
```

Comment:
Where is the corresponding precondition in the actual function?

Reply:
Yeah that's a good point. I've looked at other proofs that use the aws_hash_table_has_an_empty_slot function and none of them have a corresponding precondition in the harness. I do think this should be added, but maybe as a separate PR. I can't think of a simple, easy fix.

Reply:
Having an empty slot is implied by this line in the `hash_table_is_valid` function. Assuming that the hash-table is correctly implemented, such that all non-filled slots are marked empty, then having fewer filled slots than available slots ensures this property.
https://github.com/awslabs/aws-c-common/blob/master/source/hash_table.c#L1040
But there is not an explicit check that this is actually true. The difficulty is that this requires checking for the existance of such an actual slot, which requires a potentially expensive O(n) loop, whereas the rest of the check of O(1) time.

Reply:
Sounds good and we do have the hash_table_is_valid precondition on the function.